source/doc-pages/guides/basics/repositories.md

```diff
@@ -0,0 +1,309 @@
+there are attributes that were not expected you will get an exception.
+
+Here's an example of a struct:
+
+``` ruby
+user = user_repo[1]
+# #<ROM::Struct[User] id=1 name="Jane">
+
+user.id # 1
+user[:id] # 1
+
+user.to_hash # {:id=>1, :name=>"Jane"}
+```
+
+ROM repositories generate struct classes for you so that you don't have to define
+them.
+
+<div class="well well-small">
+There's a planned feature where you'll be able to provide namespace for your
+structs where struct class constants will be defined as opposed to anonymous classes.
+</div>
+
+## Working With Relations
+
+In ROM relations represent specific database queries that your application needs
+to execute to get its data. It is recommended to define relations explicitly as
+your own classes to achieve better encapsulation and simplify composition.
+
+A common pattern is to encapsulate specific relation methods and use those in your
+repositories rather than accessing lower-level query DSLs.
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  def listing
+    select(:id, :name, :email, :created_at).order(:name, :id)
+  end
+
+  def registered_after(timestamp)
+    where { created_at > timestamp }
+  end
+end
+
+class UserRepository < ROM::Repository::Base
+  relations :users
+
+  def new_users(timestamp)
+    users.listing.registered_after(timestamp)
+  end
+end
+
+user_repo.new_users(Time.new(2015))
+```
+
+This way you encapsulate specific queries and domain-specific relation views. Your
+application sees simple structs and has no idea how they are fetched and instantiated.
+
+## Relation Views
+
+Every method that you define in a relation represents a specific relation **view**.
+For relation views that you re-use to compose more complex relations you can
+explicitly define their structure with `view` plugin:
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  view(:listing, [:id, :name, :email, :created_at]) do
+    select(:id, :name, :email, :created_at).order(:name, :id)
+  end
+
+  def registered_after(timestamp)
+    where { created_at > timestamp }
+  end
+end
+```
+
+Thanks to this feature we can be explicit about the data structures our relations
+return which comes with the benefit of auto-mapping and better introspection
+capabilities that can be used to build even more advanced features.
+
+<div class="well">
+The view feature will be enhanced with type annotations so that it will be possible
+to automatically map database-specific types into domain-specific types.
+</div>
+
+### Base View Plugin
+
+ROM repository provides an SQL relation plugin called `base_view` which defines
+a `base` relation view for you which includes all the column names and order by
+the primary key:
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  def filter(conditions)
+    base.where(conditions)
+  end
+end
+```
+
+## Composing Relations
+
+Every repository can access multiple relations. Thanks to this you can compose
```

Comment:
Every repository can access multiple relations. Thanks to this you can compose
them however you want and get aggregates back. There's no need
to define `associations` with complicated configuration logic, you simply define
methods on your repository objects: